### PR TITLE
framework(engine): fix stale pong heartbeat causes ack time fallback

### LINKED ISSUE #5969: test(engine): unstable node-failure e2e test
### Which jobs are flaking?

Dataflow Engine e2e tests

### Which test(s) are flaking?

Node-failure-workflow

### Jenkins logs or GitHub Actions link

https://github.com/pingcap/tiflow/pull/5713#issuecomment-1162597994

```log
https://github.com/pingcap/tiflow/runs/6981383933?check_suite_focus=true
```

log details: [node-failure-workflow-logs.zip](https://github.com/pingcap/tiflow/files/8956334/node-failure-workflow-logs.zip)

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: close #5969

TODO
- [x] add unit test

### What is changed and how it works?

- When worker receives a stale heartbeat pong message from master, don't reset `lastMasterAckedPingTime`
- Use `clock.MonotonicTime` as submit time in task context instead of `time.Time`

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/framework/internal/worker/master_client.go [hunk 1]
```diff
@@ -189,7 +189,14 @@ func (m *MasterClient) HandleHeartbeat(sender p2p.NodeID, msg *frameModel.Heartb
 	if msg.IsFinished {
 		m.masterSideClosed.Store(true)
 	}
-	m.lastMasterAckedPingTime.Store(time.Duration(msg.SendTime))
+
+	lastAckTime := m.lastMasterAckedPingTime.Load()
```
Comment (liuzix) [inline engine/framework/internal/worker/master_client.go:193]: Could you add some comment here?

Comment (amyangfei) [reply engine/framework/internal/worker/master_client.go:196]: added, ptal

**Review by gozssky — APPROVED** [review]

**Review by maxshuang — APPROVED** [review]

**Pull request merged**